packages/roosterjs-content-model/lib/editor/plugins/PastePlugin/ContentModelPastePlugin.ts

```diff
@@ -0,0 +1,88 @@
+import addParser from './utils/addParser';
+import ContentModelBeforePasteEvent from '../../../publicTypes/event/ContentModelBeforePasteEvent';
+import deprecatedColorParser from './utils/deprecatedColorParser';
+import sanitizeLinks from './utils/linkParser';
+import { getPasteSource } from 'roosterjs-editor-dom';
+import { handleWordDesktop } from './WordDesktop/handleWordDesktopPaste';
+import { IContentModelEditor } from '../../../publicTypes/IContentModelEditor';
+import {
+    EditorPlugin,
+    IEditor,
+    KnownPasteSourceType,
+    PluginEvent,
+    PluginEventType,
+} from 'roosterjs-editor-types';
+
+/**
+ * Paste plugin, handles BeforePaste event and reformat some special content, including:
+ * 1. Content copied from Word
+ * (This class is still under development, and may still be changed in the future with some breaking changes)
+ */
+export default class ContentModelFormatPlugin implements EditorPlugin {
+    private editor: IContentModelEditor | null = null;
+
+    /**
+     * Construct a new instance of Paste class
+     * @param unknownTagReplacement Replace solution of unknown tags, default behavior is to replace with SPAN
+     */
+    constructor(private unknownTagReplacement: string = 'SPAN') {}
+
+    /**
+     * Get name of this plugin
+     */
+    getName() {
+        return 'ContentModelPaste';
+    }
+
+    /**
+     * The first method that editor will call to a plugin when editor is initializing.
+     * It will pass in the editor instance, plugin should take this chance to save the
+     * editor reference so that it can call to any editor method or format API later.
+     * @param editor The editor object
+     */
+    initialize(editor: IEditor) {
+        // TODO: Later we may need a different interface for Content Model editor plugin
+        this.editor = editor as IContentModelEditor;
+    }
+
+    /**
+     * The last method that editor will call to a plugin before it is disposed.
+     * Plugin can take this chance to clear the reference to editor. After this method is
+     * called, plugin should not call to any editor method since it will result in error.
+     */
+    dispose() {
+        this.editor = null;
+    }
+
+    /**
+     * Core method for a plugin. Once an event happens in editor, editor will call this
+     * method of each plugin to handle the event as long as the event is not handled
+     * exclusively by another plugin.
+     * @param event The event to handle:
+     */
+    onPluginEvent(event: PluginEvent) {
+        if (!this.editor || event.eventType != PluginEventType.BeforePaste) {
+            return;
+        }
+
+        const ev = event as ContentModelBeforePasteEvent;
+        if (!ev.domToModelOption) {
+            return;
+        }
+        const pasteSource = getPasteSource(event, false);
+        switch (pasteSource) {
+            case KnownPasteSourceType.WordDesktop:
+                handleWordDesktop(ev);
```

Comment:
So this content will be processed by both old and new paste plugin, right? Is there any issue with that?

Reply:
My idea is to use this instead of having the functionality in the old plugin. Same for other Paste sources

Reply:
So we can't enable this plugin until it can cover all functionalities of old plugin?

If that is the plan, I think it is fine.